packages/google_maps_flutter/google_maps_flutter_web/test/test_driver/shapes_integration.dart

```diff
@@ -190,6 +190,32 @@ void main() {
       expect(polygon.get('strokeColor'), '#c0ffee');
       expect(polygon.get('strokeOpacity'), closeTo(1, _acceptableDelta));
     });
+
+    testWidgets('Handle Polygons with holes', (WidgetTester tester) async {
+      final polygons = {
+        Polygon(
+          polygonId: PolygonId('BermudaTriangle'),
+          points: [
+            LatLng(25.774, -80.19),
+            LatLng(18.466, -66.118),
+            LatLng(32.321, -64.757),
+          ],
+          holes: [
+            [
+              LatLng(28.745, -70.579),
+              LatLng(29.57, -67.514),
+              LatLng(27.339, -66.668),
+            ],
+          ],
+        ),
+      };
+
+      controller.addPolygons(polygons);
+
+      expect(controller.polygons.length, 1);
+      expect(controller.polygons, contains(PolygonId('BermudaTriangle')));
+      expect(controller.polygons, isNot(contains(PolygonId('66'))));
```

Comment:
Is there any way that we can test that points within the hole are *outside* of the polygon?

Here's a JS example:

* https://developers.google.com/maps/documentation/javascript/examples/poly-containsLocation

And this is the Dart API that you might be able to use:

* https://pub.dev/documentation/google_maps/latest/google_maps_geometry/Poly/containsLocation.html

(IIRC the Poly class is not exposed directly from the platform_interface package, so you might need to use the `google_maps` lib from the tests directly.

Reply:
I added the test to shape_integratin.dart